File: src/server/Gateway/SocialMediaApp.Aggregator/Models/Posts/PostListDto.cs
using SocialMediaApp.Aggregator.Models.Posts.Comments;
using SocialMediaApp.Aggregator.Models.Posts.Likes;
using SocialMediaApp.Aggregator.Models.Users;

namespace SocialMediaApp.Aggregator.Models.Posts
{
    public class PostListDto
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public UserListDto User { get; set; }
        public string Description { get; set; }
        public string Location { get; set; }
        public DateTime CreateDate { get; set; }
        public ICollection<LikeListDto> Likes { get; set; }
        public ICollection<CommentListDto> Comments { get; set; }
    }
}

File: src/client/user/src/models/posts/PostListDto.ts
import { UserListDto } from '@shared/models/users/UserListDto'
import { PostLikeListDto } from './PostLikeListDto'
import { PostCommentListDto } from './PostCommentListDto'

export class PostListDto {
  constructor(init: Partial<PostListDto>) {
    Object.assign(this, init)
    this.createDate = new Date(init.createDate)
    this.comments = this.comments.map((c) => new PostCommentListDto(c))
  }

  id: number
  userId: number
  user: UserListDto
  description: string
  location: string
  createDate: Date
  updatedAt: Date
  imageUrl: string
  likes: PostLikeListDto[]
  comments: PostCommentListDto[]
}
